orchagent/portsorch.cpp

```diff
@@ -2754,6 +2757,9 @@ bool PortsOrch::removeVlan(Port vlan)
         SWSS_LOG_ERROR("Failed to remove non-empty VLAN %s", vlan.m_alias.c_str());
         return false;
     }
+    /* notify aclorch to unbind vlan from acl table */
```

Comment:
any reason we need unbind vlan before remove vlan from HW?  What if the remove_vlan API failed later? See LAG case as well.

Reply:
“any reason we need unbind vlan before remove vlan from HW?”
----Sorry, I did not consider the remove_vlan API failure before.
I just thought it was better to notify port change before remove acl table group from HW. In addition, if we want to unbind, it may be better to call set_vlan_attribute API with attribute SAI_VLAN_ATTR_INGRESS_ACL setting to NULL, so it maybe better to unbind before remove vlan from HW.
But, it doesn't call set_vlan_attribute API with attribute SAI_VLAN_ATTR_INGRESS_ACL setting to NULL when unbind acl table, so we need not to unbind vlan before remove vlan from HW. In the future, the set_vlan_attribute may be called for making AclTable::bind and AclTable::unbind symmetric.(When I modified this code, I found a comment "// TODO: make bind/unbind symmetric" before AclTable::bind, I guessed it might need to call set_vlan_attribute API with attribute SAI_VLAN_ATTR_INGRESS_ACL setting to NULL, so I put PORT_CHANGE notification before remove vlan from HW ) 

“What if the remove_vlan API failed later?”
----If the remove_vlan API failed, then return false, doVlanTask() will not erase this task from "consumer.m_toSync". It will execute removeVlan() until return true, so notify(PORT_CHANGE) will call many times. But unbind() Will only be called once because port will be erased from ACLtable::portSet when calling unbind() for the first time.
On the HW, vlan still exists, but there is no acltable bound on this vlan.

So, do I need to put PORT_CHANGE notification before "removeAclTableGroup(vlan);" and after remove vlan from HW ?
I'll do this in the next commit.